surefire-api/src/main/java/org/apache/maven/surefire/api/stream/AbstractStreamEncoder.java

```diff
@@ -112,17 +113,17 @@ public void encodeString( CharsetEncoder encoder, ByteBuffer result, String stri
     {
         String nonNullString = nonNull( string );
 
-        int counterPosition = result.position();
+        int counterPosition = ( (Buffer) result ).position();
```

Comment:
`position` method is also in the Buffer class. Maybe we can directly declare such type.

Reply:
No, see https://github.com/apache/maven-surefire/pull/332#discussion_r564955600. Furthermore, the call `encoder.encode( wrap( nonNullString ), result, true );` requires `result` to be a `ByteBuffer` again.